manager-bundle/src/ContaoManager/Plugin.php

```diff
@@ -230,10 +228,17 @@ public function getExtensionConfig($extensionName, array $extensionConfigs, Plug
                 return $this->handlePrependLocale($extensionConfigs, $container);
 
             case 'framework':
+                $extensionConfigs = $this->checkMailerTransport($extensionConfigs, $container);
+                $extensionConfigs = $this->addDefaultMailer($extensionConfigs, $container);
+
                 if (!isset($_SERVER['APP_SECRET'])) {
                     $container->setParameter('env(APP_SECRET)', $container->getParameter('secret'));
                 }
 
+                if (!isset($_SERVER['MAILER_URL'])) {
+                    $container->setParameter('env(MAILER_URL)', $this->getMailerUrl($container));
+                }
```

Comment:
#### Note

Symfony's standard environment variable would be `MAILER_DSN`, not `MAILER_URL`. However, since we previously supported `MAILER_URL`, I left it this way.

Reply:
What about using `MAILER_DSN` if present and fall back to `MAILER_URL` if not provided?

Reply:
Unfortunately, as @m-vo mentioned, the DSN for the Symfony Mailer Component is not compatible with the one from Swiftmailer. So we can't fall back directly.

Reply:
If we no longer ship Swiftmailer, we do not have to provide `MAILER_URL` anymore IMHO.

Reply:
So should we just force users to change their env variable and the DSN accordingly when upgrading to 4.10?

Reply:
We can keep the logic to remain backwards compatible. Then we generate both `env(MAILER_URL)` and `env(MAILER_DNS)` from the parameters.yml file.

Reply:
Right and then check for 

```php
isset($_SERVER['MAILER_URL'])
```

[here](https://github.com/fritzmg/contao/blob/87ef78dc29c917b5370e25866bc3c549548d1b43/manager-bundle/src/ContaoManager/Plugin.php#L355-L359) and use one or the other.

Reply:
No, wait, that still does not solve the problem 😁 . If someone defined a `MAILER_URL`, it will still be incompatible with Symfony Mailer.

Reply:
Yes, that is why we will generate both `MAILER_URL` and `MAILER_DNS`. And we will use `MAILER_DNS` in our `config.yml` of course.

Reply:
But we can't generate it from the `parameters.yml` then. If someone uses the `MAILER_URL` env, then the `parameters.yml` will not contain anything. I think we have to do it as I described here: https://github.com/contao/contao/pull/1829#issuecomment-643997977

Reply:
You are right.